### PR TITLE
[utils] fix component log functions introduced at 06bbead

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Fix the component log functions of `CLog` which I added at 06bbead (#12676).

## Motivation and Context
fix a crash

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
@MilhouseVH

File: xbmc/utils/log.cpp [hunk 1]
```diff
@@ -85,11 +85,14 @@ void CLog::LogFunction(int loglevel, const char* functionName, const char* forma
...
-  if (g_advancedSettings.CanLogComponent(component))
+  if (g_advancedSettings.CanLogComponent(component) && IsLogLevelLogged(loglevel))
   {
+    std::string fNameStr;
+    if (functionName && functionName[0])
+      fNameStr.assign(functionName).append(": ");
     va_list va;
     va_start(va, format);
-    CLog::LogFunction(loglevel, functionName, format, va);
+    LogString(loglevel, fNameStr + StringUtils::FormatV(format, va));
```
Comment (Montellese) [inline xbmc/utils/log.cpp]: Can't you call
```
CLog::LogFunction(loglevel, functionName, StringUtils::FormatV(format, va));
```
That way you don't have to duplicate the function name formatting logic.

**Comment by Rechi** [comment]
WIN-64 failure unrelated
any objections?

**Comment by Rechi** [comment]
confirmed to fix the issue in latest Milhouse build, https://forum.kodi.tv/showthread.php?tid=298462&pid=2633603#pid2633603

**Pull request merged**